torchmultimodal/models/gpt.py

```diff
@@ -475,6 +491,14 @@ def forward(
         return_attn_weights: bool = False,
         return_hidden_states: bool = False,
     ) -> TransformerDecoderOutput:
+        if attn_mask is not None and attn_mask.dim() == 2:
+            attn_mask = attn_mask[
+                None, None, :, :
+            ]  # (q_seq_len, k_seq_len) -> (b, h, q_seq_len, k_seq_len)
```

Comment:
nit: isn't it technically this?
```suggestion
            ]  # (q_seq_len, k_seq_len) -> (1, 1, q_seq_len, k_seq_len)
```

Reply:
yeah I haven't been consistent with my notations. Let me update them.